scripts/m18_submission_checker/aida_m18_submission_checker.py

```diff
@@ -0,0 +1,383 @@
+            ta1_ret_val = 1
+            type_count = type_count + 1
+        #if ttls were found in both the NIST/ directory and a subdirectory of NIST/, mark as TA_1a and have the ta_1a do this check too
+        if ta1_ret_val > -1 and type_count > 1 :        #use > and not == for scaleability? idk dont @ me c:
+            logging.warning("WARNING: Subdirectories found at the same level of .ttl files in the NIST directory.")
+            ta1_ret_val = 0
+        return ta1_ret_val
+    elif dot_count > 1 and dot_count < 4:
+        return dot_count
+    else:
+        logging.error("Cannot figure out the TA level. Please ensure archive name is properly titled per NIST guidelines.")
+        return -1
+
+
+
+def validate_archive_filetype():
+    """
+    Checks the archive file type. If it is not a valid filetype, the program will
+    print an error message and quit the script.
+    Valid filetypes: .tgz, .tar.gz, or .zip.
+
+    :return: True if valid filetype and successfully set the global FILETYPE variable, False otherwise
+    :rtype: bool
+    """
+    global FILETYPE
+
+    if ARCHIVE_FILE_NAME.endswith('.tar.gz') or ARCHIVE_FILE_NAME.endswith('.tgz'):
+        FILETYPE = 0
+    elif ARCHIVE_FILE_NAME.endswith('.zip'):
+        FILETYPE = 1
+    else:
+        FILETYPE = -1
+        return False
+
+    return True
+
+def main():
+    """
+
+    :returns: True if script completed successfully, False otherwise
+    :rtype: bool
+    """
+    global FILETYPE
+    global TASK_TYPES
+    global ARCHIVE_NAMES
+    global ARCHIVE_FILE_NAME
+
+    # set logging to log to stdout
+    logging.basicConfig(level='INFO', format='%(levelname)s:  %(message)s')
+
+    #check runtime arguments
+    if len(sys.argv) < 2:
+        logging.error("Missing archive file as argument.\nTo run: python3 aida_m18_submission_checker.py <path/to/archive>")
+        return False
+    if not os.path.isfile(sys.argv[1]):
+        logging.error("Argument is not a file.\nTo run: python3 aida_m18_submission_checker.py <path/to/archive>")
+        return False
+
+    ARCHIVE_FILE_NAME = str(sys.argv[1])
+
+    #ensure the archive type is a valid filetype and set the global FILETYPE value
+    if validate_archive_filetype() < 0:
+        logging.critical("Archive filetype unknown. Please use .zip, .tgz, or .tar.gz")
+        return False
+
+    #get names of items in the archive
+    try:
+        logging.info("Checking archive... this may take a moment for large files.")
+        archive = zipfile.ZipFile(ARCHIVE_FILE_NAME, 'r', allowZip64=True) if FILETYPE else tarfile.open(ARCHIVE_FILE_NAME, mode='r')
+        ARCHIVE_NAMES = archive.namelist() if FILETYPE == 1 else archive.getnames()
+    except:
+        #expected errors: (zipfile.BadZipFile, tarfile.ReadError), but catch all of them anyways
+        logging.critical("Error thrown attempting to read/open the archive. Please use 'zip' or 'tar' to create your archive.")
```

Comment:
I would be more explicit in the text of this statement letting the user know that the only valid submissions are .tar.gz, .tgz, and .zip. This will really never be thrown once you fix the small bug with the if statement above.

Reply:
the error thrown by this block is due to 
1. nothing in the archive
2. an invalid archive format (i.e. some text file renamed to .tgz)

we previously check for appropriate file extension and warn the user

Reply:
Okay, the only reason I suggested this was due to the failure in one of my test cases that I described below. It made it through your validation check when it should have failed and this was the messages I received.

Reply:
this is strange, ill test this out more.